report/3.Theory/LanguageSpecification.tex

```diff
@@ -76,6 +85,27 @@ \section{Language Specification}
 ["john", "frank"] where {value is "frank"} print
 \end{lstlisting}
 
+\textbf{Grammar for function calls}\\
+There are three different ways to call a function depending on how many parameters must be passed with the function:
+\begin{itemize}
+	\item If no parameters must be passed, then invoking the function identifier is a call in and of itself.
+	\item If one parameter must be passed, then the parameter must appear on the left-hand side of the function identifier.
+	\item If more than one parameter must be passed, the first parameter must appear on the left-hand side of the function identifier, and the rest must appear, in order, on the right-hand side of the function.
+\end{itemize}
+With this in mind, the grammar for function calls has been specified as follows:
+\begin{align*}
+	\texttt{<FunctionCall>}\to & \texttt{ identifier}\\
+	| & \texttt{ <FunCallParam> identifier [<FunCallParams>]}
+\end{align*}
+As it is not allowed to have parameters on the right-hand side of the identifier without also having one on the left-hand side, this rule cannot accurately be described with only one production.
+
+The first production shows how the function is called if it has no parameters.
+The second production shows that if the function has parameters, there must always be exactly one parameter on the left-hand side of the identifier, and optionally a sequence of parameters on the right-hand side.
+
+The non-terminal \texttt{<FunCallParam>} refers to any legal parameter for function calls, including identifiers, literals, lambda expressions, and other function calls.
+
+The non-terminal \texttt{<FunCallParams>} refers to a sequence of at least one \texttt{<FunCallParam>}, optionally enclosed in parentheses.
+
 \textbf{Loops}\\
```

Comment:
explain that loops are in fact a function just as everything else

Reply:
kinda explained in line 111, but could be more explicit

Reply:
fair